addons/crm_partner_assign/models/crm_lead.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import random
+from openerp import api, fields, models, _
+from openerp.exceptions import UserError
+from openerp.addons.base_geolocalize.models.res_partner import geo_find, geo_query_address
+
+
+class CrmLead(models.Model):
+    _inherit = 'crm.lead'
+    partner_latitude = fields.Float(string='Geo Latitude', digits=(16, 5))
+    partner_longitude = fields.Float(string='Geo Longitude', digits=(16, 5))
+    partner_assigned_id = fields.Many2one('res.partner', string='Assigned Partner', track_visibility='onchange', help="Partner this case has been forwarded/assigned to.", index=True)
+    date_assign = fields.Date(string='Assignation Date', help="Last date this case was forwarded/assigned to a partner")
+
+    def get_interested_action(self, interested):
+        try:
+            lead_channel_act = self.env.ref('crm_partner_assign.crm_lead_channel_interested_act')
+        except ValueError:
+            raise UserError(_("The CRM Channel Interested Action is missing"))
```

Comment:
if external id not found for crm channel , it's raise exception by self.env.ref. no need to use try and exception here. by default "raise_if_not_found=True" in ref() method.